argus/action_manager/windows.py

```diff
@@ -120,32 +121,85 @@ def get_installation_script(self):
         self.download_resource("windows/installCBinit.ps1",
                                r"C:\installCBinit.ps1")
 
+    def check_cbinit_installation(self):
+        """Check if Cloudbase-Init was installed successfully."""
+        LOG.info("Checking Cloudbase-Init installation.")
+
+        try:
+            python_dir = introspection.get_python_dir(
+                lambda cmd, command_type: self._client.run_command_with_retry(
+                    cmd, count=util.RETRY_COUNT,
+                    delay=util.RETRY_DELAY, command_type=command_type)[0]
+            )
+        except exceptions.ArgusError as exc:
+            LOG.warning("Could not check Cloudbase-Init installation: %s", exc)
+            self.cbinit_cleanup()
+            return False
+
+        check_cmd = r'& "{}\python.exe" -c "import cloudbaseinit"'.format(
+            python_dir)
+        try:
+            self._client.run_remote_cmd(
+                cmd=check_cmd, command_type=util.POWERSHELL)
+        except exceptions.ArgusError as exc:
+            LOG.debug("Cloudbase-Init installation failed: %s", exc)
+            self.cbinit_cleanup()
+            return False
+
+        LOG.info("Cloudbase-Init was successfully installed!")
+        return True
+
+    def cbinit_cleanup(self):
+        """Clean-up Cloudbase-Init if it's installed."""
+        LOG.info("Cleaning up Cloudbase-Init.")
+
+        try:
+            cbinit_dir = introspection.get_cbinit_dir(
+                lambda cmd, command_type: self._client.run_command_with_retry(
```

Comment:
Should I rewrite and use that `_execute()` method instead of lambda expressions?